src/ripple/protocol/impl/TxFormats.cpp

```diff
@@ -322,6 +325,71 @@ TxFormats::TxFormats()
             {sfTicketSequence, soeOPTIONAL},
         },
         commonFields);
+
+    add(jss::XChainCreateBridge,
+        ttXCHAIN_CREATE_BRIDGE,
+        {
+            {sfXChainBridge, soeREQUIRED},
+            {sfSignatureReward, soeREQUIRED},
+            {sfMinAccountCreateAmount, soeOPTIONAL},
+        },
+        commonFields);
+
+    add(jss::XChainModifyBridge,
+        ttXCHAIN_MODIFY_BRIDGE,
+        {
+            {sfXChainBridge, soeREQUIRED},
+            {sfSignatureReward, soeOPTIONAL},
+            {sfMinAccountCreateAmount, soeOPTIONAL},
+        },
+        commonFields);
+
+    add(jss::XChainCreateClaimID,
+        ttXCHAIN_CREATE_CLAIM_ID,
+        {
+            {sfXChainBridge, soeREQUIRED},
+            {sfSignatureReward, soeREQUIRED},
+            {sfOtherChainSource, soeREQUIRED},
+        },
+        commonFields);
+
+    add(jss::XChainCommit,
+        ttXCHAIN_COMMIT,
+        {
+            {sfXChainBridge, soeREQUIRED},
+            {sfXChainClaimID, soeREQUIRED},
+            {sfAmount, soeREQUIRED},
+            {sfOtherChainDestination, soeOPTIONAL},
+        },
+        commonFields);
+
+    add(jss::XChainClaim,
+        ttXCHAIN_CLAIM,
+        {
+            {sfXChainBridge, soeREQUIRED},
+            {sfXChainClaimID, soeREQUIRED},
+            {sfDestination, soeREQUIRED},
+            {sfDestinationTag, soeOPTIONAL},
+            {sfAmount, soeREQUIRED},
```

Comment:
I'm playing around with `XChainClaim`.  I see that the `Amount` represents the amount that will be recovered by the `XChainClaim`.  That seems fair.

However I'm seeing that the account submitting the `XChainClaim` has two things taken:

- The `Fee` and
- The `SignatureReward`.

Having the `Fee` taken is great.  That `Fee` is explicitly filled in by the account submitting the transaction.

However, having the `SignatureReward` disappear from the account took me by surprise.  There's nothing explicit in the transaction to indicate that those funds will be removed.  From what I've seen we like folks to explicitly state how much they are expecting to spend when they submit a transaction.

So my off-the-cuff suggestion would be that the `XChainClaim` transaction must contain a `SignatureReward` field that exactly matches the `SignatureReward` in the `XChainClaimID` that they are claiming.  That way the transaction explicitly identifies every amount that this transaction will cost.

Reply:
Well, the `XChainClaim` tx will only succeed if a quorum of attestations is present, and those attestations do specify the reward that the `XChainClaim`  deducts from the door account (in addition to the `fee` and the `amount`), so the reward deduction should not come as a surprise. 

Also the `XChainClaim` is not strictly necessary, and the transfer can occur automatically when a quorum of attestations is reached, so you could argue that this would be even more surprising, the  `amount` + `reward` being taken from the door account without the door account submitting *any* transaction.

Reply:
I have to disagree.  It is certainly obvious that _someone_ will need to pay the `SignatureReward` for the claim.  But in the case of the `XChainClaim` it's not obvious _who_ will pay the reward.

For a normal XChain transfer I see both the `Amount` and the `SignatureReward` being paid by the door account.  That makes sense to me.  The door account explicitly signed up for the job of supplying cross-chain funds.  And the `SignatureReward` is subtracted from the `Amount`.  So only the total `Amount` moves out of the door account.

However in the `XChainClaim` case, the door account is no longer paying the `SignatureReward`.  This time it's the account that submits the `XChainClaim` transaction.  By submitting that transaction they know they are signing up to pay a `Fee`, but they may not know that they, not the door, will be expected to pay the `SignatureReward`.

Potentially the actual problem is that the door should be paying the `SignatureReward` for the `XChianClaim` (and subtracting that reward from the `Amount`), just like it does for a normal XChain transfer.  I'd be fine with that too.  But I find having the submitter of the `XChainClaim` being charged for the `SignatureReward` without explicitly saying so to be surprising.

Reply:
I do see your point. I actually think that the reward is always taken from the account which created the claimID to begin with (whether there is a `XChainClaim` or not), but I'm not 100% sure.

Reply:
I agree, I thought the signature reward was taken out at the `XChainCreateClaimID` transaction, and not the `XChainClaim` transaction

Reply:
@mvadari I don't think so, I see the reward being deducted either when a quorum of attestation is reached (for the case when a dest account is specified) or when the `XChainClaim` transaction is executed.

Reply:
> Potentially the actual problem is that the door should be paying the SignatureReward for the XChainClaim

Well, we want the amounts locked on the door in the locking chain to match the amounts issued by the door on the issuing chain. Since in the current design, only the `amount` (not the `reward`) is locked in the door account on the locking chain, it makes sense that the reward is *not* issued by the door on the issuing chain. I think that is why the reward is always paid by the account which reserved the `claimID`.

Reply:
Well, I was wrong and @greg7mdp is correct.  I had misunderstood some behaviors.  Going back to earlier explorations I'm seeing consistent behavior that the `SignatureReward` is always paid by the creator of the `XChainClaimID`.  And the transaction that creates the `XChainClaimID` does explicitly specify exactly what `SignatureReward` is being offered for providing the attestations.

Sorry for the noise.  I got this one wrong.